src/js/components/order.js

```diff
@@ -10,6 +11,53 @@ import PanelGroup from 'react-bootstrap/lib/PanelGroup';
 import Row from 'react-bootstrap/lib/Row';
 import Table from 'react-bootstrap/lib/Table';
 
+class AmountField extends React.Component {
```

Comment:
Extract this component out of Order is exactly right.
We could do more to rename it to ```NumberInput```, and separate it to another file in terms of reusing NumberInput somewhere.